l_display.py

```diff
@@ -0,0 +1,295 @@
+            "black",
+            #           "ochre"
+            #           "peach"
+            #           "ruby"
+            "olive",
+            "violet",
+            #           "fawn"
+            #           "lilac"
+            "gold",
+            "chocolate",
+            #           "mauve"
+            #           "cream"
+            "crimson",
+            "silver",
+            #           "rose"
+            "azure",
+            #           "lemon"
+            #           "russet"
+            "grey",
+            "purple",
+            #           "white"
+            "pink",
+            "orange",
+            "blue",
+        ]
+        for elem in root.findall(".//{http://www.music-encoding.org/ns/mei}l"):
+            facs = elem.get("facs").strip("#")
+            i += 1
+
+            for zone in zones:
+
+                if (
+                    zone.get("{http://www.w3.org/XML/1998/namespace}id")
+                    and zone.get("{http://www.w3.org/XML/1998/namespace}id") == facs
+                ):
+                    ulx, uly, lrx, lry = (
+                        int(int(zone.get("ulx")) * scale) // 4,
+                        int(int(zone.get("uly")) * scale) // 4,
+                        int(int(zone.get("lrx")) * scale) // 4,
+                        int(int(zone.get("lry")) * scale) // 4,
+                    )
+                    draw.rectangle(
+                        [(ulx, uly), (lrx, lry)], outline=colours[i % len(colours)]
+                    )
+
+        photo = ImageTk.PhotoImage(image)
+
+        # Update the image label
+        image_label.config(image=photo)
+        image_label.image = photo
+
+        # Clear the text frame
+        for widget in text_frame.winfo_children():
+            widget.destroy()
+
+        # Update the text fields
+        text_fields.clear()
+        for i, elem in enumerate(
+            root.findall(".//{http://www.music-encoding.org/ns/mei}l")
+        ):
+            label = ttk.Label(
+                text_frame, text=f"Line {i+1} ({colours[i%len(colours)]}):"
+            )
+            label.grid(row=i, column=0, padx=5, pady=5)
+            # Create a colored frame to wrap the Entry widget
+            text_field = ttk.Entry(text_frame, width=50)
+            text_field.insert(0, elem.text)
+            text_field.grid(row=i, column=1, padx=5, pady=5)
+            text_fields.append(text_field)  # Function to increment the index
+        index_label.config(text=f"Page: {index:04d}")
+        scale_label.config(text=f"Scale: {scale_slider.get():.2f}")
+
+    def increment_index():
+        """
+        Increment the index to display the next page.
+
+        This function is called when the user clicks the "+" button or presses the right arrow key.
+        """
+        nonlocal index
+        while True:
+            index += 1
+            image_file = f"liber_{index:04d}.jpg"
+            image_path = os.path.join(image_folder, image_file)
```

Comment:
You're doing a lot of the same string manipulation over and over again ... do you need to? For example, here you do it to check that the file exists, and then you do it again immediately after when you call `update_image`